### PR TITLE
#164796896 user should be able to follow and unfollow eachother

### PR DESCRIPTION
**Pull request by Raymond-Osy**
#### Description
- This is an API endpoint for a user to be able to follow and unfollow another user

#### Type of change
- New feature (non-breaking change that adds functionality)

#### How Has This Been Tested?
- [x] Integration test
- [x] Unit test

#### Checklist:
- [x] Style guide

#### PT
[#164796896](https://www.pivotaltracker.com/story/show/164796896)

#### Snap shots
User following postman test
Screenshot 2019-04-16 at 8 54 12 PM

User unfollowing postman test
Screenshot 2019-04-16 at 8 54 36 PM

### CODE REVIEW

File: src/controllers/follower.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        })
+      );
+    }
+    if (findUser.id === user.id) {
+      return res.status(403).json(
+        errorResponseFormat({
+          status: 'fail',
+          message: 'Sorry, you can not unfollow yourself'
+        })
+      );
```
Comment (ssanusi) [inline src/controllers/follower.js]: @Raymond-Osy you can implement follow and unfollow as a toggle functionality. click to follow. once you click again you unfollow that way you can use findOrCreate you already have for following functionality

File: src/routers/follower.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// import { Router } from 'express';
```
Comment (ssanusi) [inline src/routers/follower.js]: @Raymond-Osy remember to remove this code if they are not needed

File: src/routers/index.js [hunk 3]
```diff
@@ -58,6 +60,10 @@ router.get('/auth/google/callback', passport.authenticate('google', { failureRed
 
 router
   .post('/signup', checkBody, createUser);
+router.post('/login', checkFields, login);
+router.post('/followers/:followerId/follow', Authenticator.verifyToken, followUser);
+router.delete('/followers/:followerId/unfollow', Authenticator.verifyToken, unfollowUser);
```
Comment (ssanusi) [inline src/routers/index.js]: @Raymond-Osy if you implement as toggle functionality you don't need  unfollow endpoint

**Review by ssanusi — COMMENTED** [review]
@Raymond-Osy nice work I see you considered most edge cases but you could have one endpoint to implement follow and unfollow

File: src/controllers/follower.js (hunk 1 above)
Comment (micah-akpan) [reply src/controllers/follower.js]: @Raymond-Osy. Nice work.  you can modify the error response to only return  `status` and `data` in the body.

Comment (Raymond-Osy) [reply src/controllers/follower.js]: Thanks, @ssanusi for the feedback, I never thought of implementing it that way. I have implemented these changes.

Comment (Raymond-Osy) [reply src/controllers/follower.js]: Thanks @micah-akpan for spotting it, I've adjusted it accordingly

File: src/routers/follower.js (hunk 2 above)
Comment (Raymond-Osy) [reply src/routers/follower.js]: Sure, thanks

File: src/routers/index.js (hunk 3 above)
Comment (Raymond-Osy) [reply src/routers/index.js]: Yes, thanks for the idea...it makes sense. Implemented.

**Comment by micah-akpan** [comment]
@Raymond-Osy please Squash the commits. Allowable number of commits on a PR is 1 to 5 commits. Thanks

**Pull request closed without merging**